### PR TITLE
Bump nelmio/alice to ~2.0

### PR DESCRIPTION
**Pull request by teohhanhui**

### CODE REVIEW

**Comment by akovalyov** [comment]
@teohhanhui it is a little bit more complicated than to bump version in composer since we rely on `Nelmio\Alice\Loader\Yaml`, which was either renamed to https://github.com/nelmio/alice/blob/master/src/Nelmio/Alice/Fixtures/Parser/Methods/Yaml.php or removed (not sure if I followed 2.0 changes). Do you have a willing to apply the corresponding changes in this PR? :)
And when this PR is merged, @PedroTroller could tag new 0.6 release I believe.

**Comment by teohhanhui** [comment]
Am I correct to think that `nelmio/alice` can actually be moved to `require-dev` and `suggest`?

**Comment by teohhanhui** [comment]
FYI `@BeforeBackground` tag is broken with Behat 3. For example, `Knp\FriendlyContexts\Context\AliceContext::loadAlice` is run once for each scenario.

**Comment by teohhanhui** [comment]
ping @akovalyov

**Comment by BenoitLeveque** [comment]
:+1:

**Comment by gl3n** [comment]
:+1:

File: src/Knp/FriendlyContexts/Alice/Fixtures/Loader.php [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function instantiateFixtures(array $fixtures)
+    {
+        parent::instantiateFixtures($fixtures);
+
+        /** @var \Nelmio\Alice\Fixtures\Fixture $fixture */
```
Comment (PedroTroller) [inline src/Knp/FriendlyContexts/Alice/Fixtures/Loader.php]: Can you remove this comment ?

File: src/Knp/FriendlyContexts/Alice/Fixtures/Loader.php [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        foreach ($fixtures as $fixture) {
+            /** @var \Nelmio\Alice\Fixtures\PropertyDefinition $property */
```
Comment (PedroTroller) [inline src/Knp/FriendlyContexts/Alice/Fixtures/Loader.php]: Same here ?

File: composer.json [hunk 3]
```diff
@@ -12,13 +12,16 @@
...
         "fzaninotto/faker": "~1.0",
-        "nelmio/alice": "~1.7",
         "guzzle/guzzle": "~3.7"
     },
     "require-dev": {
         "phpspec/phpspec": "2.0.*@dev",
         "doctrine/orm": "dev-master",
-        "knplabs/phpspec-welldone-extension": "dev-master"
+        "knplabs/phpspec-welldone-extension": "dev-master",
+        "nelmio/alice": "~2.0"
```
Comment (PedroTroller) [inline composer.json]: Why require dev ? It can't works without alice...

**Comment by teohhanhui** [comment]
@PedroTroller Comments fixed.

File: composer.json [hunk 4]
```diff
@@ -20,6 +20,9 @@
         "doctrine/orm": "dev-master",
         "knplabs/phpspec-welldone-extension": "dev-master"
     },
+    "suggest": {
+        "nelmio/alice": "To use Alice context"
```
Comment (Einenlum) [inline composer.json]: Useless now.

**Comment by teohhanhui** [comment]
@Einenlum Fixed.

**Pull request merged**